commit 4a2a24bcf670920076446c53bdc2dc2f35552c6d (upstream)
Author: Christian Mollekopf <chrigi_1@fastmail.fm>
Date:   Fri Apr 13 18:06:21 2012 +0200

    Check the actual id of the resource to decide if it is the owner.
    
    Otherwise other resources, acting as client, are mistakenly detected as owner of the item.
    (cherry picked from commit 21c49789128fcf56b01b5ba97d4830c7c837c6fb)

diff --git a/server/src/akonadiconnection.cpp b/server/src/akonadiconnection.cpp
index 256692d..6ea97ef 100644
--- a/server/src/akonadiconnection.cpp
+++ b/server/src/akonadiconnection.cpp
@@ -294,7 +294,7 @@ void AkonadiConnection::setResourceContext(const Resource& res)
 
 bool AkonadiConnection::isOwnerResource(const PimItem& item) const
 {
-  if ( resourceContext().isValid() )
+  if ( resourceContext().isValid() && item.collection().resourceId() == resourceContext().id() )
     return true;
   // fallback for older resources
   if ( sessionId() == item.collection().resource().name().toUtf8() )
@@ -304,7 +304,7 @@ bool AkonadiConnection::isOwnerResource(const PimItem& item) const
 
 bool AkonadiConnection::isOwnerResource(const Collection &collection) const
 {
-  if ( resourceContext().isValid() )
+  if ( resourceContext().isValid() && collection.resourceId() == resourceContext().id() )
     return true;
   if ( sessionId() == collection.resource().name().toUtf8() )
     return true;
